pkg/pg/collectors.go

```diff
@@ -12,6 +13,47 @@ import (
 	"github.com/jackc/pgx/v5/pgxpool"
 )
 
+// Helper function reformat and emit a single metric
+func EmitMetric(state *agent.MetricsState, metric metrics.MetricDef, value any) error {
+	entry, err := metric.AsFlatValue(value)
+	if err == nil {
+		state.AddMetric(entry)
+	} else {
+		return err
+	}
+	return nil
+}
+
+type Number interface {
+	~int64 | ~float64
+}
+
+type MetricMapping[T Number] struct {
+	Current  *T
+	Previous *T
+	Metric   metrics.MetricDef
+}
+
+// Helper function to calculate the delta and emit a number of metrics
+// The mapping takes the current value, the previous value and the metric to emit the metric to
+// If either current or previous is nil, the metric is skipped
+// In some cases, the delta can be negative (e.g. if the counter has been reset)
```

Comment:
Do we want to skip all metrics where the delta can be negative? This assumption is based on the fact that all metrics are mono-tonically increasing which I don't believe is true.